### ISSUE TITLE
Multiple loading indicators in monument details view

### ISSUE THREAD

**Issue by mdmohsin7** [AOSSIE-Org/Monumento#178]
There are multiple loading indicators which appear in the monument details view of the app. Ideally there should only be one indicator for both the components.

<img width="788" alt="Screenshot 2024-09-01 at 3 12 15 PM" src="https://github.com/user-attachments/assets/46bed0a7-9e33-4d52-a9ec-726245d59ef4">

**Comment by ChandraDevs1** [comment]
Can I Work On It 
Please Assign it

**Comment by mdmohsin7** [comment]
> Can I Work On It Please Assign it

Assigned!

**Comment by karthik14344** [comment]
Hey @mdmohsin7, I can see no activity going on this issue. Can you please assign me this issue? let's get it done.

**Comment by mdmohsin7** [comment]
Assigned! Pls go ahead @karthik14344

**Comment by kareemabdeen** [comment]
Iam dn with #186 pr, Can you please assign it to me ?  @mdmohsin7

**Comment by kareemabdeen** [comment]
There are two ways to solve this issue, 
**Approach 1: Single Loading Indicator**
Create a _isLoading function to check if any bloc is loading and display a single CircularProgressIndicator in the center using a Stack.

**Approach 2: Shimmer Effect**
Use the redacted package to wrap widgets, enabling shimmer placeholders (enabled: _isLoading()) for a better loading experience.

I suggest going with the second approach as it is better for the user experience 
I need your opinion to go ! @mdmohsin7

**Comment by khushi-hura** [comment]
Hey @mdmohsin7 can I work on this issue, please assign!

**Comment by mdmohsin7** [comment]
Let's go with the first approach now otherwise shimmer on a single screen would not look that good. @kareemabdeen feel free to open another issue to replace circular indicators with shimmers in all the screens where it makes sense

Assigning this issue to @khushi-hura for now as @kareemabdeen already has an issue assigned!

**Comment by khushi-hura** [comment]
@mdmohsin7 can you please specify the screen in which the two circular indicators are being appeared?

**Comment by aayaan758** [comment]
hey @mdmohsin7 can u assign this to me,i can work on this

**Comment by khushi-hura** [comment]
@aayaan758 I am working on this!

**Comment by aayaan758** [comment]
@khushi-hura oh okay ...

**Comment by khushi-hura** [comment]
hey @mdmohsin7 I want your help regarding firebase issues after setup, my account is not a billing account and it is asking to setup billing account before using storage as per new policies of firebase!
Please help!
thanks in advance :)

**Comment by mdmohsin7** [comment]
> hey @mdmohsin7 I want your help regarding firebase issues after setup, my account is not a billing account and it is asking to setup billing account before using storage as per new policies of firebase! Please help! thanks in advance :)

Hi, you'll have to add your payment details for now, dw you won't get charged. In coming weeks we will migrate over to pocketbase.io or supabase.com or appwrite.io

**Comment by khushi-hura** [comment]
> > hey @mdmohsin7 I want your help regarding firebase issues after setup, my account is not a billing account and it is asking to setup billing account before using storage as per new policies of firebase! Please help! thanks in advance :)
> 
> Hi, you'll have to add your payment details for now, dw you won't get charged. In coming weeks we will migrate over to pocketbase.io or supabase.com or appwrite.io

I have successfully added my payment details but they are asking for prepayment of 1000 INR :(

**Comment by khushi-hura** [comment]
hey @mdmohsin7 is there anything we can do on the firebase setup issue as of now or should I wait for the organization to switch to supabase.com or appwrite.io...??

**Comment by kareemabdeen** [comment]
I can take over this issue if @khushi-hura is still encountering problems, since we have already spent a lot of time addressing it. @mdmohsin7

**Comment by kareemabdeen** [comment]
@khushi-hura, could you please provide more details about the issue you're currently facing? I'd like to help you complete your contribution until @mdmohsin7 decides on our migration strategy from Firebase.

**Comment by khushi-hura** [comment]
Hi @kareemabdeen yes I am still facing issues with using storage in firebase.
For using storage, firebase is asking to setup a billing account and to setup a billing account, I am required to setup autopay and they are asking for a pre payment of 1000 INR. Since @mdmohsin7  informed that we are moving to supabase.com or appwrite.io in coming weeks, so I was waiting for more guidance as I think that autopay could be risky as it won't request me to make a payment, it will just deduct money from my bank account!

**Comment by kareemabdeen** [comment]
I think you don't need to use the billing account, as we are using Firestore, not Cloud Storage. 
@khushi-hura

**Comment by khushi-hura** [comment]
Ohh!
But after doing the setup correctly and cross checking all the functionalities, the code couldn't reach to specified storage path in firebase @kareemabdeen 
Should I share logs if u say?

Also I think that in September, firebase has been updated everything to its billing account 

Could u plz make a setup and let me know if everything is working fine without setting up a billing account? Then I will try doing it again with your help.

**Comment by kareemabdeen** [comment]
@khushi-hura Have you enabed Firestore?
it shouldnt need a bank account  
Check this simple video and try to follow these steps. I hope this helps.
https://www.youtube.com/watch?v=T6B6sj6ZfqU

**Comment by khushi-hura** [comment]
hey @kareemabdeen thanks for helping me out. There was something wrong with the setup, now working fine with the app!
btw I cannot find double loading indicators in the specified file in this issue, if you could specify file it would be great @mdmohsin7

**Comment by kareemabdeen** [comment]
Hey @khushi-hura , glad to hear the setup issue is resolved and the app is working fine now! You can check the **MonumentDetailsViewMobile** file, the issue with the double loading indicators should be there. Let me know if you need further assistance!

**Comment by khushi-hura** [comment]
hey @mdmohsin7 @kareemabdeen is this what fixes the issue?
The screen now displays 1 progress indicator 

![Screenshot_20250101-124316](https://github.com/user-attachments/assets/c34a07e4-968b-48fd-8c9e-32e9ad43225b)

**Comment by kareemabdeen** [comment]
Hi @khushi-hura,

It seems the issue occurs when navigating to the details page for the **first time**, where the initial state shows two loading indicators instead of one. To help identify and resolve this, please follow these steps:

1. Navigate to the details page for the first time.


2. Observe whether the initial state displays two loading indicators. Ideally, there should only be one centered on the page.

**Comment by kareemabdeen** [comment]
@khushi-hura  Could you please provide more details about state now for this issue ?

**Comment by khushi-hura** [comment]
Yes @kareemabdeen I am sorry, I was out for new year. Can I please get 3 more days?
Just will raise the PR by 10th Jan or if I get stuck somewhere Will let u know by tomorrow!

**Comment by kareemabdeen** [comment]
@khushi-hura 
I was just making sure that you aren’t facing any new problems or something like that

**Comment by khushi-hura** [comment]
hey @kareemabdeen I just identified that the code is managing _isLoading boolean variable to ensure the state and at the same time the bloc is also managing the state, both are using circularProgressIndicators() which is resulting in 2 of them.
I tried fixing this. If that is the point I will raise the PR.
Correct me if I am wrong.

**Comment by kareemabdeen** [comment]
@khushi-hura 
Yes, you're right! Please include a screen recording showing the before and after changes with each pull request to facilitate the review process by @mdmohsin7

**Comment by khushi-hura** [comment]
ok thanks @kareemabdeen

**Comment by kareemabdeen** [comment]
You're welcome, @khushi-hura

**Comment by dharmikpathak** [comment]
i can fix it , 
assign me

**Comment by kareemabdeen** [comment]
Kindly grant me permission to work on this issue to resolve it promptly, as it has been taking considerable time to address.
@mdmohsin7

**Comment by lakshyasarve** [comment]
@mdmohsin7 Could you please assign this issue to me?

**Comment by ShashwatXD** [comment]
is this issue solved?

**Comment by lakshyasarve** [comment]
@mdmohsin7 Could you please assign this issue to me?
